### PR TITLE
Support for file retrieval

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java [hunk 1]
```diff
@@ -102,16 +108,77 @@ public void shouldCallInterceptorChainProcessor() throws Exception {
...
+    public void shouldPassEntityResponseStrategyToProcessor() throws Exception {
+
+        generator.run(
+                restRamlWithQueryApiDefaults().with(
+                        resource("/path2")
+                                .with(httpAction(GET).withDefaultResponseType())
+                ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, emptyMap()));
+
+        Class<?> resourceClass = compiler.compiledClassOf(BASE_PACKAGE, "resource", "DefaultPath2Resource");
```
Comment (mapingo) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java]: can be final

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java [hunk 2]
```diff
@@ -102,16 +108,77 @@ public void shouldCallInterceptorChainProcessor() throws Exception {
...
+    public void shouldPassEnvelopePayloadEntityResponseStrategyNameToProcessor() throws Exception {
+
+        generator.run(
+                restRamlWithQueryControllerDefaults().with(
+                        resource("/path")
+                                .with(httpAction(GET).withDefaultResponseType())
+                ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, emptyMap()));
+
+        Class<?> resourceClass = compiler.compiledClassOf(BASE_PACKAGE, "resource", "DefaultPathResource");
```
Comment (mapingo) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java]: can be final

File: generators/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsImplementationGenerator.java [hunk 3]
```diff
@@ -188,14 +175,36 @@ private TypeSpec generateFor(final Resource resource, final Optional<String> com
                 .addField(FieldSpec.builder(FileInputDetailsFactory.class, FILE_INPUT_DETAILS_FACTORY_FIELD)
                         .addAnnotation(Inject.class)
                         .build());
+        return classSpec;
```
Comment (mapingo) [inline generators/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsImplementationGenerator.java]: classSpec can be inlined

File: generators/rest-adapter-file-service/src/main/java/uk/gov/justice/services/adapter/rest/processor/response/FileStreamReturningResponseStrategy.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    public Response responseFor(final String action, final Optional<JsonEnvelope> result) {
+
+        try {
+            final FileReference fileReference = fileRetriever.retrieve(fileIdFrom(result)).orElseThrow(NotFoundException::new);
+            return responseStrategyHelper.responseFor(action, result,
+                    jsonEnvelope -> status(OK)
+                            .entity(fileReference.getContentStream())
+                            .build());
+        } catch (FileServiceException e) {
+            return null;
```
Comment (mapingo) [inline generators/rest-adapter-file-service/src/main/java/uk/gov/justice/services/adapter/rest/processor/response/FileStreamReturningResponseStrategy.java]: Should something other than null be returned here?

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java (hunk 1 above)
Comment (jaceko) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java]: done

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java (hunk 2 above)
Comment (jaceko) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java]: done

File: generators/rest-adapter-file-service/src/main/java/uk/gov/justice/services/adapter/rest/processor/response/FileStreamReturningResponseStrategy.java (hunk 4 above)
Comment (jaceko) [inline generators/rest-adapter-file-service/src/main/java/uk/gov/justice/services/adapter/rest/processor/response/FileStreamReturningResponseStrategy.java]: done

**Pull request merged**